rust/relay/src/server.rs

```diff
@@ -131,35 +155,126 @@ where
         sender: SocketAddr,
         now: Instant,
     ) -> Result<()> {
-        let Ok(message) = self.decoder.decode_from_bytes(bytes)? else {
-            tracing::trace!("received broken STUN message from {sender}");
-            return Ok(());
-        };
+        if tracing::enabled!(target: "wire", tracing::Level::TRACE) {
+            let hex_bytes = hex::encode(bytes);
+            tracing::trace!(target: "wire", r#"Input::client("{sender}","{hex_bytes}")"#);
+        }
 
-        tracing::trace!("Received message {message:?} from {sender}");
+        // De-multiplex as per <https://www.rfc-editor.org/rfc/rfc8656#name-channels-2>.
+        match bytes.first() {
+            Some(0..=3) => {
+                let Ok(message) = self.decoder.decode_from_bytes(bytes)? else {
+                    tracing::trace!(target: "relay", "received broken STUN message from {sender}");
```

Comment:
Sure. The judgement I usually apply for `warn` is "Do I want to wake an ops person at 3am if this happens?".

This really only happens if someone is spamming our UDP socket with random data. Yeah, it might be a DoS but I think realistically, it is probably just random internet traffic probing ports.

Reply:
Haha, gotcha, I was thinking of `warn` because it could be a meaningful log when debugging why a connection isn't working.